### ISSUE TITLE
C++ API documentation question

### ISSUE THREAD

**Issue by sandeep-ps** [mlpack/mlpack#3369]
This question is related to the JOSS paper review https://github.com/openjournals/joss-reviews/issues/5026 and criteria checklist entry `Documentation - Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?`

Since this is primarily a C++ library, are the library's C++ API methods documented somewhere? I could see similar documentation for other programming language bindings (e.g., https://www.mlpack.org/doc/stable/python_documentation.html), but not the ones for C++.

**Comment by rcurtin** [comment]
Thanks for pointing this out. :+1: For a long time, we used Doxygen to provide C++ class documentation, but we actually found it to be difficult to maintain and actually uglier than simply browsing the source itself!  So, the recommendation now is to simply browse the source code for what you are looking for, and the methods in the source themselves are more than sufficient for usage:

For instance, here is the source code for the `KMeans` class; every method (and every parameter to every method) is fully documented:

https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/kmeans/kmeans.hpp

We realized while answering this that this fact isn't pointed out in the documentation page---so I added a paragraph to indicate this: https://github.com/mlpack/mlpack.org/commit/b6a32121ee863e20cc2a20d2155426d7eee4edab

I hope this clarifies the concern. :+1:
